include/download.h

```diff
@@ -54,6 +58,7 @@ class Download {
 	std::string fn;
 	std::string url_;
 	DlStatus download_status;
+	const char* msg;
```

Comment:
This is initialized from either `curl_easy_strerror()` or `strerror()`. The former [currently returns one of a few static strings](https://github.com/curl/curl/blob/76140ecfde258b9d397ad660c7dc671f0b0ff9ce/lib/strerror.c#L56-L361), but [the docs make no guarantees about that](https://curl.haxx.se/libcurl/c/curl_easy_strerror.html). The latter [explicitly states](https://www.man7.org/linux/man-pages/man3/strerror.3.html) that later calls might modify the existing string.

All in all, it's better to store a copy of the string using [`std::string`](https://en.cppreference.com/w/cpp/string/basic_string). Can you please update the code?

Reply:
sure, will do!